File: 3747-maximum-difference-between-adjacent-elements-in-a-circular-array/3747-maximum-difference-between-adjacent-elements-in-a-circular-array.cpp
class Solution {
public:
    int maxAdjacentDistance(vector<int>& nums) {
        int maxi=0;
        for(int i=1;i<nums.size();++i){
            maxi=max(maxi,abs(nums[i-1]-nums[i]));
        }
        maxi=max(maxi,abs(nums[0]-nums[nums.size()-1]));
        return maxi;
    }
};

File: 3747-maximum-difference-between-adjacent-elements-in-a-circular-array/3747-maximum-difference-between-adjacent-elements-in-a-circular-array.java
class Solution {
    public int maxAdjacentDistance(int[] nums) {
        int maxi=0;
        for(int i=1;i<nums.length;++i){
            maxi=Math.max(maxi,Math.abs(nums[i-1]-nums[i]));
        }
        maxi=Math.max(maxi,Math.abs(nums[0]-nums[nums.length-1]));
        return maxi;
 
    }
}